### ISSUE TITLE
Mark make deps as non-explictly installed

### ISSUE THREAD

**Issue by Arzumify** [Jguer/yay#2207]
### Is your feature request related to a problem? Please describe.

No

### Describe the solution you'd like

Marking make dependencies as non-explicitly installed via pacman -D --asdeps

### Describe alternatives you've considered

Yay having custom markings for make dependencies

### Additional context

Because it's annoying when the process gets interrupted and your computer gets filled with unused make dependencies

**Comment by Jguer** [comment]
@Tracker-Friendly make deps are marked as dependencies in the DB, `yay -Yc` does not propose you to clean them?

**Comment by smolx** [comment]
@Jguer, Although issue-author didn't mention version, I dare to guess that it can be a regression bug in git-version after #2196. In case of "the process gets interrupted"(at a certain moment), installed packages won't be marked as deps. Maybe, as a workaround, yay should install packages with --asdeps option by default and after that apply an "explicit" reason only if needed.

**Comment by Arzumify** [comment]
Not to mention, compatible with default pacman purge (pacman -Qdtq)

**Comment by Jguer** [comment]
@smolx There is a risk there. If packages get installed as explicit, the user will want to remove unneeded but there will be none, wasteful but not dramatic. If the packages get installed as dependencies but then they fail to get set as explicit, when the user goes to remove unneeded, he'll either end up without the package he wanted or with a broken system

**Comment by smolx** [comment]
@Jguer Yeah, you are right. Then I think there is only one way left to do it right: install package and set reason with one command. But I'm sure there will be nuances too and I don't have the expertise to fully evaluate it.
